jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/JobParameterValidationListener.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.jsr352.massindexing;
+
+import javax.batch.api.BatchProperty;
+import javax.batch.api.listener.AbstractJobListener;
+import javax.inject.Inject;
+
+import org.hibernate.search.exception.SearchException;
+import org.hibernate.search.jsr352.logging.impl.Log;
+import org.hibernate.search.util.logging.impl.LoggerFactory;
+
+import static org.hibernate.search.jsr352.massindexing.impl.util.ValidationUtil.validateCheckpointInterval;
+import static org.hibernate.search.jsr352.massindexing.impl.util.ValidationUtil.validatePositive;
+
+/**
+ * Listener for job parameters validations before the start of a job execution.
+ *
+ * @author Mincong Huang
+ */
+public class JobParameterValidationListener extends AbstractJobListener {
+
+	private static final Log log = LoggerFactory.make( Log.class );
+
+	@Inject
+	@BatchProperty(name = MassIndexingJobParameters.CHECKPOINT_INTERVAL)
+	private String checkpointIntervalStr;
+
+	@Inject
+	@BatchProperty(name = MassIndexingJobParameters.ROWS_PER_PARTITION)
+	private String rowsPerPartitionStr;
+
+	public JobParameterValidationListener() {
+	}
+
+	JobParameterValidationListener(String checkpointIntervalStr, String rowsPerPartitionStr) {
```

Comment:
This constructor, having default access modifier, is designed for the unit-tests. I want to keep this "shortcut" for instance. I'll remove it once we have a better way to inject the job parameters via constructor or setter methods, which will be handled in ticket [HSEARCH-2632](https://hibernate.atlassian.net/browse/HSEARCH-2632).